pkg/apiserver/rest/apis/v1/types.go

```diff
@@ -1088,17 +1090,62 @@ type LoginResponse struct {
 	RefreshToken string             `json:"refreshToken,omitempty"`
 }
 
-// DetailUserResponse is the detail user info for the response
-type DetailUserResponse struct {
-	Name  string `json:"name"`
-	Alias string `json:"alias,omitempty"`
-	Email string `json:"email"`
-}
-
 // DexConfigResponse is the response of dex config
 type DexConfigResponse struct {
 	ClientID     string `json:"clientID"`
 	ClientSecret string `json:"clientSecret"`
 	RedirectURL  string `json:"redirectURL"`
 	Issuer       string `json:"issuer"`
 }
+
+// DetailUserResponse is the response of user detail
+type DetailUserResponse struct {
+	UserBase
+	Projects []ProjectUserBase `json:"projects"`
+}
+
+// ProjectUserBase project user base
+type ProjectUserBase struct {
+	Name     string `json:"name"`
+	Alias    string `json:"alias"`
+	UserRole string `json:"userRole"`
+}
+
+// CreateUserRequest create user request
+type CreateUserRequest struct {
+	Name  string `json:"name" validate:"checkname"`
+	Alias string `json:"alias,omitempty" validate:"checkalias" optional:"true"`
+	Email string `json:"email" validate:"checkemail"`
+	// TODO: add password validation
+	Password string `json:"password"`
```

Comment:
add the password validate

Reply:
Not sure of the validation, is there any rule for it?